core/conn/odb/src/odb.c

```diff
@@ -4156,18 +4156,17 @@ static void sigcatch(int sig)
     exit ( EX_SIGNAL );
 #else 
     if ( tn == 1 ) { /* single threaded */
-        tclean( 0 );
         gclean();
-        exit( EX_SIGNAL );
     } else {
         for ( i = 0 ; i < tn ; i++ ) {
-            if ( !pthread_kill(thid[i], 0) ) {  /* If this thread is alive... */
+            if ( pthread_kill(thid[i], 0) ) {  /* If this thread is alive... */
```

Comment:
pthread_kill returns 0 when the thread exists. So, earlier code ( ! pthread_kill ...) is a valid one

Reply:
Yes, I misunderstood it.